internal/executor/job/testfixtures.go

```diff
@@ -0,0 +1,84 @@
+package job
+
+import (
+	"sync"
+
+	"github.com/armadaproject/armada/pkg/api"
+)
+
+// TestJobRunStateStore Just wraps JobRunStateStore but allows tests to set the initial state
+type TestJobRunStateStore struct {
+	jobRunStateStore *JobRunStateStore
+}
+
+func NewTestJobRunStateStore(initialJobRuns []*RunState) *TestJobRunStateStore {
```

Comment:
Maybe I'm being dim but why do we need a whole new struct here  Can't we just have a `NewJobRunStateStoreWithInitialState` function in `job_run_state_store.go` that returns a a `JobRunStateStore` with the state set?

Reply:
Yes I didn't start like this - but this was the end result.

I'll just do that